### PR TITLE
provide API to easily hash test data when constructing ClusteredTestD…

### PR DESCRIPTION
**Pull request by karottenreibe**
…etails

Addresses issue https://jira.cqse.eu/browse/TS-24353

- [x] Changes are tested adequately
- [x] Agent's README.md updated in case of user-visible changes
- [x] CHANGELOG.md updated
- [x] Present new features in [N&N](https://wiki.cqse.eu/pages/viewpage.action?pageId=689566)
- [x] [TGA Tutorial](https://docs.teamscale.com/tutorial/setting-up-tga-java) updated
- [x] [TIA Tutorial](https://docs.teamscale.com/tutorial/tia-java/) updated

Please respect the vote of the [Teamscale bot](https://demo.teamscale.com) or flag irrelevant findings as tolerated or false positives. If you feel that the Teamscale config needs adjustment, please state so in a comment and discuss this with your reviewer.

### CODE REVIEW

**Comment by karottenreibe** [comment]
I fixed the architecture problem on master, so after merging, it should be gone :-)

**Review by DreierF — CHANGES_REQUESTED** [review]
Awesome 👏 

There are just some compile errors now because of the new constructor overload

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ We use [semantic versioning](http://semver.org/):
 
 # 19.0.0
 - [fix] option parsing errors were not logged in rare cases
+- [feature] _tia-client_: add API to hash test data when creating ClusteredTestDetails
```
Comment (DreierF) [inline CHANGELOG.md:15]: This should appear under next release

**Comment by karottenreibe** [comment]
sorry, I'm ready for a vacation :P

**Pull request merged**